### PR TITLE
[Universal] Restricts on Graphics Tests Projects

### PR DESCRIPTION
**Pull request by lukaschod**
# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: TestProjects/UniversalGraphicsTest_Foundation/Assets/Test/Editor/FoundationEditorTests.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            var path = AssetDatabase.GUIDToAssetPath(guid);
+
+            // We only care what is in assets folder
+            if (!path.StartsWith("Assets"))
+                continue;
+
+            LightingSettings lightingSettings = AssetDatabase.LoadAssetAtPath<LightingSettings>(path);
+            //Assert.IsFalse(lightingSettings.realtimeGI, $"Lighting Setting at path {path} has realtime GI enabled. Foundation project should not have realtime GI");
+            //Assert.IsFalse(lightingSettings.bakedGI, $"Lighting Setting at path {path} has realtime GI enabled. Foundation project should not have baked GI");
```
Comment (ellioman) [inline TestProjects/UniversalGraphicsTest_Foundation/Assets/Test/Editor/FoundationEditorTests.cs:47]: These tests not working or?

Comment (lukaschod) [reply TestProjects/UniversalGraphicsTest_Foundation/Assets/Test/Editor/FoundationEditorTests.cs:47]: Yes, so it is something to figure out. As seems like foundation tests are depending on reflection probes generated by GI. Maybe we should only restrict realtime gi, not sure

File: com.unity.testing.urp/Scripts/Editor/UniversalProjectAssert.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            if (!path.StartsWith("Assets"))
+                continue;
+
+            LightingSettings lightingSettings = AssetDatabase.LoadAssetAtPath<LightingSettings>(path);
+            Assert.IsFalse(lightingSettings.realtimeGI, $"Lighting Setting ({path}) has realtime GI enabled. {projectName} project should not have realtime GI");
+            Assert.IsFalse(lightingSettings.bakedGI, $"Lighting Setting ({path}) has baked GI enabled. {projectName} project should not have baked GI");
+        }
+    }
+
+    public static void AllShadersAreNot(string projectName, ShaderPathID shaderPathID)
```
Comment (ellioman) [inline com.unity.testing.urp/Scripts/Editor/UniversalProjectAssert.cs:112]: Can we rename this function to something a bit clearer?

**Review by ellioman — COMMENTED** [review]
Looks good so far. Just need to get those TODO's done

**Review by ellioman — APPROVED** [review]
Looks good 👍

**Review by erikabar — APPROVED** [review]

**Comment by lukaschod** [comment]
URP_Terrain on iPhone_Metal_Standalone_il2cpp_Linear on version 2021.2 
UniversalGraphicsTests.035_Shader_TerrainShaders

URP_Terrain on Win_DX11_playmode_XR_mono_Linear on version 2021.2
?

URP_Terrain on OSX_Metal_playmode_mono_Linear on version 2021.2
?

URP_Terrain on Android_Vulkan_Standalone_il2cpp_Linear on version 2021.2
Vulkan thread Connection issues

URP_Lighting on iPhone_Metal_Standalone_il2cpp_Linear on version 2021.2
UniversalGraphicsTests.017_Lighting_Scene_DirectionalBaked

URP_Lighting on Win_DX11_playmode_XR_mono_Linear on version 2021.2 
?

URP_Foundation on iPhone_Metal_Standalone_il2cpp_Linear on version 2021.2
UniversalGraphicsTests.143_LightCookies_ManyLights_deferred

URP_Foundation on OSX_Metal_playmode_mono_Linear on version 2021.2
?

**Pull request merged**